### PR TITLE
change: reshape Artifacts into data frame in ExperimentsAnalytics

### PR DESCRIPTION
**Pull request by yzhu0**
Reshape Artifacts into data frame in ExperimentsAnalytics

https://sim.amazon.com/issues/AML-71634

### CODE REVIEW

File: src/sagemaker/analytics.py [hunk 1]
```diff
@@ -516,6 +520,36 @@ def _reshape_metrics(self, metrics):
                     out["{} - {}".format(metric_name, stat_type)] = stat_value
         return out
 
+    def _reshape_input_artifacts(self, input_artifacts):
```
Comment (jerrypeng7773) [inline src/sagemaker/analytics.py:523]: why can't we just have _reshape_artifacts() which can be applied to both input/output-artifacts, so we dont need two unit tests over this too =)

File: src/sagemaker/analytics.py [hunk 2]
```diff
@@ -463,6 +465,8 @@ def __init__(
         self._sort_order = sort_order
         self._metric_names = metric_names
         self._parameter_names = parameter_names
+        self._input_artifact_names = input_artifact_names
+        self._output_artifact_name = output_artifact_names
```
Comment (leoroy) [inline src/sagemaker/analytics.py:469]: You might want to add comments for these 2 arguments in the above section as well.

File: src/sagemaker/analytics.py (hunk 1 above)
Comment (danabens) [reply src/sagemaker/analytics.py:523]: +1 would eliminate some redundancy

Comment (yzhu0) [reply src/sagemaker/analytics.py:523]: thats true,

File: src/sagemaker/analytics.py (hunk 2 above)
Comment (yzhu0) [reply src/sagemaker/analytics.py]: added

**Comment by danabens** [comment]
looks good but not clear why unit test check is failing

**Comment by nadiaya** [comment]
Pylint has failed:
```
src/sagemaker/analytics.py:455:0: C0301: ·[1mLine too long (119/100)·[0m (·[1mline-too-long·[0m)
src/sagemaker/analytics.py:457:0: C0301: ·[1mLine too long (119/100)·[0m (·[1mline-too-long·[0m)
```

File: src/sagemaker/analytics.py [hunk 3]
```diff
@@ -450,6 +452,11 @@ def __init__(
             sagemaker_session (sagemaker.session.Session): Session object which manages interactions
                 with Amazon SageMaker APIs and any other AWS services needed. If not specified,
                 one is created using the default AWS configuration chain.
+            input_artifact_names(dict optional):The input artifacts for the experiment. Examples of
+             input artifacts are datasets, algorithms, hyperparameters, source code, and instance
```
Comment (ChuyangDeng) [inline src/sagemaker/analytics.py:456]: Could you correct the indentations in the docstring?

Comment (yzhu0) [reply src/sagemaker/analytics.py]: fixed

**Pull request merged**